src/components/Autocomplete/AutocompleteResults/AutocompleteResults.tsx

```diff
@@ -39,23 +40,35 @@ const DefaultRenderResults: RenderResults = ({ sections }) =>
 
 export default function AutocompleteResults(props: AutocompleteResultsProps) {
   const { children = DefaultRenderResults } = props;
-  const { sections, isOpen, getMenuProps, getItemProps, closeMenu } =
-    useContext(CioAutocompleteContext);
-
+  const {
+    query,
+    sections,
+    isOpen,
+    isFocused,
+    getMenuProps,
+    getItemProps,
+    closeMenu,
+    advancedParameters,
+  } = useContext(CioAutocompleteContext);
+  const { displayNoResultsMessage } = advancedParameters || {};
   const hasResults = sections && sections.some((section) => section?.data?.length);
 
   const menuProps = {
     ...getMenuProps(),
   };
 
   let content;
+
   if (isOpen && hasResults) {
     content = (
       <>
         <CloseIcon onClick={() => closeMenu()} />
         {typeof children === 'function' ? children({ sections, getItemProps }) : children}
       </>
     );
+  } else if (query?.length && isFocused && displayNoResultsMessage) {
```

Comment:
This wrongly shows the "no results" message 

- When the results are still loading from the API while you're typing
- When you type something then click escape
- When you select a suggestion

Reply:
What if we check that the API response is empty to decide whether to show the message or not?

Reply:
> What if we check that the API response is empty to decide whether to show the message or not?

We allow "custom section" so we I don't think we can just check the api response by itself. I will see if there is another way to address the issue when the api is loading.

Reply:
Can we do no results per section? 🤔 

So the message appears if there's no Search Suggestion side by side with product suggestions for example

Reply:
We probably can add add this configuration under each section and do it that way.

Reply:
Made the changes to do it per section, I think that addresses all the issues.